pandas/core/arrays/datetimes.py

```diff
@@ -141,6 +141,10 @@ def f(self):
             result = fields.get_date_name_field(values, field, reso=self._creso)
             result = self._maybe_mask_results(result, fill_value=None)
 
+        if self._hasna:
+            result = fields.get_date_field(values, field, reso=self._creso)
+            result = self._maybe_mask_results(result, fill_value=None, convert="float64")
```

Comment:
not sure I follow, why add this?

Reply:
I think I added that because of the _field_accessor in timedeltas but I am not sure why I put it there. I will remove it for now.